File: 67-add-binary/67-add-binary.cpp
class Solution {
public:
    string addBinary(string a, string b)
    {
        int m=a.size(),n=b.size();
        int i=m-1,j=n-1;
        int carry=0;
        string s="";
        while(i>=0 || j>=0)
        {
            int sum=carry;
            if(i>=0)
            {
                sum+=a[i]-'0';
                i--;
            }
            if(j>=0)
            {
                sum+=b[j]-'0';
                j--;
            }
            
            s+=(sum%2+'0');
            carry=sum/2;
           
        }
        
        if(carry!=0)
        {
            s+=(carry+'0');
        }
        
        reverse(s.begin(),s.end());
        
        return s;
        
    }
};

File: 67-add-binary/67-add-binary.java
class Solution {
    public String addBinary(String a, String b)
    {
        int m=a.length(),n=b.length();
        int i=m-1,j=n-1;
        int carry=0;
        StringBuilder s=new StringBuilder();
        
        while(i>=0 || j>=0)
        {
            int sum=carry;
            if(i>=0)
            {
                sum+=a.charAt(i)-'0';
                i--;
            }
            if(j>=0)
            {
                sum+=b.charAt(j)-'0';
                j--;
            }
            
           s.append(sum%2);
            carry=sum/2;
           
        }
        
        if(carry!=0)
        {
            s.append(carry);
        }
        
       return s.reverse().toString(); 
        
    }
}